index.ts

```diff
@@ -32,17 +36,48 @@ server.tool(
     to: z.string().email().describe("Recipient email address"),
     subject: z.string().describe("Email subject line"),
     content: z.string().describe("Plain text email content"),
+    // If sender email address is not provided, the tool requires it as an argument
+    ...(!senderEmailAddress
+      ? {
+          from: z
+            .string()
+            .email()
+            .nonempty()
+            .describe(
+              "Sender email address. This must be provided explicitly provided by the end user and cannot be empty. Do not automatically populate this."
+            ),
+        }
+      : {}),
   },
-  async ({ to, subject, content }) => {
-    const data = await resend.emails.send({
-      from: "me@yoko.dev", // TODO - you should replace this with your own email! Verify domain here on Resend https://resend.com/domains
+  async ({ from, to, subject, content }) => {
+    const fromEmailAddress = from ?? senderEmailAddress;
+
+    // Type check on from, since "from" is optionally included in the arguments schema
+    // This should never happen.
+    if (typeof fromEmailAddress !== "string") {
```

Comment:
This should never happen, but `from` is `unknown` given the optional inclusion of the argument